components/automate-cli/cmd/chef-automate/pullAndGenerateConfig.go

```diff
@@ -395,6 +396,24 @@ func (p *PullConfigsImpl) fetchInfraConfig() (*ExistingInfraConfigToml, error) {
 			sharedConfigToml.Postgresql.Config.RootCA = pgRootCA
 		}
 		sharedConfigToml.Postgresql.Config.EnableCustomCerts = true
+	} else {
+		externalOsDetails := getExternalOpensearchDetails(a2ConfigMap)
+		if externalOsDetails != nil {
+			sharedConfigToml.ExternalDB.Database.Opensearch.OpensearchDomainName = externalOsDetails.OpensearchDomainName
+			sharedConfigToml.ExternalDB.Database.Opensearch.OpensearchInstanceURL = externalOsDetails.OpensearchInstanceURL
+			sharedConfigToml.ExternalDB.Database.Opensearch.OpensearchRootCert = externalOsDetails.OpensearchRootCert
+			sharedConfigToml.ExternalDB.Database.Opensearch.OpensearchSuperUserName = externalOsDetails.OpensearchSuperUserName
+			sharedConfigToml.ExternalDB.Database.Opensearch.OpensearchSuperUserPassword = externalOsDetails.OpensearchSuperUserPassword
+		}
```

Comment:
Do we have AWS Snapshot details also needed?
`aws_os_snapshot_role_arn`
`os_snapshot_user_access_key_id`
`os_snapshot_user_access_key_secret`